reactive_graph/src/traits.rs

```diff
@@ -639,3 +639,199 @@ pub fn panic_getting_disposed_signal(
+        self.try_with_value(Self::Value::clone)
+    }
+}
+
+/// A variation of the [`Write`] trait that provides a signposted "always-non-reactive" API.
+/// E.g. for [`StoredValue`](`crate::owner::StoredValue`).
+pub trait WriteValue: Sized + DefinedAt {
+    /// The type of the signal's value.
+    type Value: Sized + 'static;
+
+    /// Returns a non-reactive write guard, or `None` if the signal has already been disposed.
+    #[track_caller]
+    fn try_write_value(&self) -> Option<UntrackedWriteGuard<Self::Value>>;
+
+    /// Returns a non-reactive write guard.
+    ///
+    /// # Panics
+    /// Panics if you try to access a signal that has been disposed.
+    #[track_caller]
+    fn write_value(&self) -> UntrackedWriteGuard<Self::Value> {
+        self.try_write_value().unwrap_or_else(unwrap_signal!(self))
+    }
+}
+
+/// A variation of the [`Update`] trait that provides a signposted "always-non-reactive" API.
+/// E.g. for [`StoredValue`](`crate::owner::StoredValue`).
+pub trait UpdateValue: DefinedAt {
+    /// The type of the value contained in the signal.
+    type Value;
+
+    /// Updates the value by applying a function, non-reactively, returning the value returned by that function,
+    /// or `None` if the signal has already been disposed.
+    #[track_caller]
+    fn try_update_value<U>(
+        &self,
+        fun: impl FnOnce(&mut Self::Value) -> U,
+    ) -> Option<U>;
+
+    /// Updates the value by applying a function, non-reactively, returning the value returned by that function.
+    ///
+    /// # Panics
+    /// Panics if you try to update a signal that has been disposed.
+    #[track_caller]
+    fn update_value<U>(&self, fun: impl FnOnce(&mut Self::Value) -> U) -> U {
+        self.try_update_value(fun)
+            .unwrap_or_else(unwrap_signal!(self))
+    }
+}
+
+impl<T> UpdateValue for T
+where
+    T: WriteValue,
+{
+    type Value = <Self as WriteValue>::Value;
+
+    #[track_caller]
+    fn try_update_value<U>(
+        &self,
+        fun: impl FnOnce(&mut Self::Value) -> U,
+    ) -> Option<U> {
+        let mut guard = self.try_write_value()?;
+        Some(fun(&mut *guard))
+    }
+}
+
+/// A variation of the [`Set`] trait that provides a signposted "always-non-reactive" API.
+/// E.g. for [`StoredValue`](`crate::owner::StoredValue`).
+pub trait SetValue: DefinedAt {
+    /// The type of the value contained in the signal.
+    type Value;
+
+    /// Updates the value by replacing it, non-reactively.
+    ///
+    /// If the signal has already been disposed, returns `Some(value)` with the value that was
+    /// passed in. Otherwise, returns `None`.
+    #[track_caller]
+    fn try_set_value(&self, value: Self::Value) -> Option<Self::Value>;
+
+    /// Updates the value by replacing it, non-reactively.
+    ///
+    /// # Panics
+    /// Panics if you try to access a signal that has been disposed.
+    #[track_caller]
+    fn set_value(&self, value: Self::Value) {
```

Comment:
`set_value()` definitely doesn't need to panic, here, as it's not even returning a value.

Reply:
Same reason as `update_value`. Updated